pkgdb/src/buildenv/realise.cc

```diff
@@ -117,10 +117,27 @@ createEnvironmentStorePath(
   std::vector<RealisedPackage> & pkgs,
   nix::StorePathSet &            references,
   std::map<nix::StorePath, std::pair<std::string, resolver::LockedPackageRaw>> &
-    originalPackage )
+                                     originalPackage,
+  const std::optional<std::string> & serviceConfigPath )
 {
   /* build the profile into a tempdir */
   auto tempDir = nix::createTempDir();
+  if ( serviceConfigPath.has_value() )
+    {
+      try
+        {
+          std::filesystem::copy_file( *serviceConfigPath,
+                                      std::filesystem::path( tempDir )
+                                        / "service-config.yaml",
+                                      std::filesystem::copy_options::none );
```

Comment:
> Disclaimer: I'm reviewing this on a phone and haven't looked closely at the code, but I wonder if storing that at the top directory is the right place, and further if there aren't other files that we will want to embed in the environment going forward. By way of example we might find it makes sense to include a copy of the entire .flox directory, similar to the way that nix profile embeds the manifest.json file in the profiles it creates (particularly when we come to implement inheritance features). If and when we do that we may find it makes sense to group such files together rather than dropping them in the top level directory.

Would `${FLOX_ENV}/etc` be suitable for this particular file?

Or is there a grouping (and, the hard part, a name) of future files that would make sense?